### DISCUSSION TITLE
Uploading python code via API but using an IDE like VS code instead of the IDLE3 program

### DISCUSSION THREAD

**Discussion by EJKT** [bitcraze/discussions#133] [Q&A]
Hey!

I just got the Crazyflie 2.1 and the flow deck yesterday and have it all set up. I can upload python code via the API (with the crazy radio) but only using the IDLE application for python. I would like to be able to upload the same code but using VS code. There must be a way to do this as the only issue continues to be library imports (cflib to be specific). The cflib folder has dependencies within the python directory (where all Python-related source files lie), meaning when I move it to a more preferable location it does not function. Any solutions? FYI, I'm uploading the very basic example code for the motion commander - see below.

```
import sys

import logging
import time

# sys.path.insert(0,'C:\Users\Ethan\AppData\Local\Programs\Python\Python39\Lib\site-packages\cflib')
import cflib.crtp
from cflib.crazyflie.syncCrazyflie import SyncCrazyflie
from cflib.positioning.motion_commander import MotionCommander

URI = 'radio://0/80/2M'

# Only output errors from the logging framework
logging.basicConfig(level=logging.ERROR)


if __name__ == '__main__':
    # Initialize the low-level drivers (don't list the debug drivers)
    cflib.crtp.init_drivers(enable_debug_driver=False)

    with SyncCrazyflie(URI) as scf:
        # We take off when the commander is created
        with MotionCommander(scf) as mc:
            print('Taking off!')
            time.sleep(1)

            # mc.up(0.2)
            # time.sleep(1)

            mc.forward(0.5)
            time.sleep(1)

            # mc.down(0.2)
            # time.sleep(1)

            # We land when the MotionCommander goes out of scope
            print('Landing!')
```

Now, this aside I have a question about the code itself: when the MotionCommander is opened, it automatically signals for take-off and when it ends it automatically signals for a landing. This is in theory useful but means that when I give commands like mc.up(0.2) it rises 0.2 from its current altitude; so I'm if there is a way to fully encode all steps including take-off and landing (I have seen there exist functions for both of these actions but am unsure of how to implement). Finally, is there a way to output readings from the sensors real-time during the flight induced by python code? Apparently, the client cannot be used for this since the crazy radio only takes 1 input maximum, so maybe printing them in the console is the only option. But even if so, how do I do that and is it related to logging?

Also, I'm a 9th grade with a solid base in coding and Arduino-related projects but some of the frameworks such as APIs are still slightly unfamiliar so if I use any terminology incorrectly please do not hesitate to correct me - I want to learn as fast as possible!

Sorry for the chunky message!

Kind Regards,

Ethan Thiebaut

**Comment by knmcguire** [comment]
Hi!

I saw that in https://github.com/orgs/bitcraze/discussions/134 that you answered managed to solve this already by reading the documentation. For the next time that you open up a discussion, could you indicate in that same discussion itself if you already managed to solve the problem and provide the solution ? That would be easier for us to handle the support questions and prevents us from writing a long answer when it is no longer necessary.
